api/src/db/models/opportunity_models.py

```diff
@@ -451,3 +457,80 @@ class OpportunityChangeAudit(ApiSchemaTable, TimestampMixin):
         BigInteger, ForeignKey(Opportunity.opportunity_id), primary_key=True, index=True
     )
     opportunity: Mapped[Opportunity] = relationship(Opportunity)
+
+
+class OpportunityCompetition(ApiSchemaTable, TimestampMixin):
```

Comment:
Could we put these in a `competition_models.py` file - just want to avoid the opportunity file getting really long, and there are going to be a lot of additional tables we need for this. We can also name the tables without prefixing with `Opportunity` - they're all sorta a part of it, but separate enough